src/tdywy.c

```diff
@@ -918,14 +926,16 @@ PetscErrorCode TDyWYResidual(TS ts,PetscReal t,Vec U,Vec U_t,Vec R,void *ctx) {
       sign = PetscSign(TDyADotBMinusC(&(tdy->N[dim*f]),&(tdy->X[dim*f]),
                                       &(tdy->X[dim*c]),dim));
       for(j=0; j<nv; j++) {
-        div += sign*tdy->vel[dim*(f-fStart)+j]*wgt*tdy->V[f];
+        div += sign*tdy->vel[nv*(f-fStart)+j]*wgt*tdy->V[f];
```

Comment:
Why is `wgt(= 1/nv)` used in computation of `div`? Also, is `rho` missing in the computation of `div` and `r[c]` ([below](https://github.com/TDycores-Project/TDycore/pull/28/files#diff-55c021d81f73d9728a8c237b533b9688R933))?

Reply:
Yes, I have taken rho = constant for now and therefore cancels out of both sides of the equation. I don't want to introduce more complexity before I figure out our current problems. However, we need that and some sort of pressure dependence (Glenn suggested a linear model in PFLOTRAN) so we can solve when fully saturated. Current tests simply don't hit that state.

Reply:
The weight bit appears to be a mistake as well, nice catch.